package.json

```diff
@@ -24,6 +24,8 @@
   "scripts": {
     "start": "(cd examples/main && (path-exists node_modules || npm i) && npm run start-local)",
     "build": "./scripts/build.sh && ./scripts/collect-metrics.sh",
+    "preversion": "npm run test",
+    "postversion": "git push && git push --tags",
```

Comment:
Is this git/npm integration field?

Q: Are we going to have different setups in different repos? 

Nit: I prefer to push tag by tag rather than all tags, in case I have some cruft locally.

Reply:
> Tags make it easier to list releases in e.g. the GitHub GUI.

Sure. There is even additional integration that can be done to make sure the CHANGELOG updates makes it into github version notes as well.

> In deck.gl tags are handled by lerna. luma.gl is moving towards a monorepo as well. 

I think that is the way to do it. Define 2-3 profiles (monorepo/lerna, more advanced unirepo, simpler repo) and try to keep the documentation and tooling for each category somewhat consistent. 

Sounds like this represents an improved direction for the middle category.

> The other smaller libraries can opt-in, but I don't think it's a big deal.

Yes that makes sense of course.

Though there are also other mid-size repos in the "react-map-gl weight class" such as nebula.gl and new as yet unpublished repos. Identifying these categories of repos, stating which repos belong to which category and being consistent across them would be a good thing.